client/me/developer/main.jsx

```diff
@@ -26,8 +28,62 @@ class Developer extends Component {
 		recordTracksEvent( 'calypso_me_is_dev_account_toggled', {
 			enabled: isDevAccount ? 1 : 0,
 		} );
+
+		const cookies = cookie.parse( document.cookie );
+
+		if ( isDevAccount && ! [ 'completed', 'dismissed' ].includes( cookies.developer_survey ) ) {
+			const noticeId = 'developer-survey';
+
+			this.props.successNotice(
```

Comment:
What do you think about using the Banner component instead? Alongside accepting dismissPreferenceName as a prop, it aligns more with the other survey prompts in Calypso, such as the Themes feedback survey. (this might be a design decision though!)